custom_components/spotcast/__init__.py

```diff
@@ -302,15 +305,18 @@ def start_casting(call):
             client.transfer_playback(device_id=spotify_device_id, force_play=force_playback)
         else:
             play(client, spotify_device_id, uri, random_song, repeat, shuffle, position)
-        if shuffle or repeat:
-            time.sleep(3)
+        if shuffle or repeat or start_volume_pct <= 100:
+            if start_volume_pct <= 100:
+                _LOGGER.debug("Setting volume to %d", start_volume_pct)
+                time.sleep(2)
+                client.volume(volume_percent=start_volume_pct, device_id=spotify_device_id)
             if shuffle:
                 _LOGGER.debug("Turning shuffle on")
-                time.sleep(2)
+                time.sleep(5)
```

Comment:
Why 5?

Reply:
I removed the initial sleep(3) and added it here to the sleep(2). Reason being that the volume adjustment needs to take place sooner to affect the initial output. I'm not entirely sure that it matters to wait that long to set these but it also these aren't as time sensitive.

Reply:
Right, I think it is ok with 3 seconds in that case in both of the cases